src/App.js

```diff
@@ -0,0 +1,254 @@
+
+import "./App.css";
+import { useState } from "react";
+import Papa from "papaparse";
+
+const Header = (props) => {
+  return (
+    <header>
+      <h1>{props.title}</h1>
+      <p>{props.message}</p>
+    </header>
+  );
+}
+
+function FileUpload(props) {
+  return(
+    <div>
+      <p>File name: {props.fileName}</p>
+      <p>File type: {props.fileType}</p>
+      <p>Size in bytes: {props.fileSize}</p>
+      <p>
+        Last modified date:{' '}
+        {props.fileDate}
+      </p>
+    </div>
+   )
+}
+
+const Instructions = (props) => {
+  return (
+    <div className='image-comparison-tool'>
+      <h2>Upload a CSV file to start comparing images</h2>
+      <p>This tool will display a side-by-side snapshot of two URLs from CSV.</p>
+      <h3>How to use</h3>
+      <ol>
+        <li>Prepare your CSV file with the following columns: <code>oldUrl</code>, <code>newUrl</code>, <code>status</code>.</li>
+        <li>Upload your CSV file.</li>
+        <li>You can view the results by clicking on the <b>Previous</b> and <b>Next</b> buttons.</li>
+        <li>To add a status or test result, click on <b>Accept</b> button to mark the data as passed. Otherwise, click on the <b>Reject</b> button.</li>
+        <li>Once you are done, download the CSV file.</li>
+      </ol>
+      
+    </div>
+  )
+}
+const Navigation = (props) => {
+  return (
+    <div className="navigation">
+      <button onClick={props.onPrev}>{'\u2B05'} Previous</button>
+      <button onClick={props.onModalClose}>Close</button>
+      <button onClick={props.onNext}>Next {'\u27A1'}</button>    
+    </div>
+  )
+}
+
+const Download = (props) => {
+  return (
+    <div className="download">
+      <button id="downloadBtn" onClick={props.onDownload} value="download">Download</button>
+    </div>
+  )
+}
+
+const Preview = (props) => {
+  return (
+    <div id="preview-all">
+      <div className="column">
+        <p>
+          <strong>PAGE {props.activeData + 1}:</strong>
+          <a href={props.oldUrl}> {props.oldUrl}</a>
+        </p>
+        <iframe title="preview-0" className="preview" src={'https://image.thum.io/get/width/1024/fullpage/allowJPG' + props.oldUrl} width="100%" height="500" />
```

Comment:
I have used the API from Thum.io because it gives more flexibility and it can capture the full page.

**Potential issue we could get from this API:**
1.  It's using a free version. There could be limitations on the number of URLs.

**Why I did not use `mShots`:** 
1.  I cannot find a way to capture the full page. The images were cropped. 
2. If you enter a new URL for the first time, you need to wait and navigate to the URL a second time to see the generated image. I will then need to add a check to reload if the URL was redirected to `https://s0.wp.com/mshots/v1/default`

Reply:
Ahh, yeah, this is a bummer... 😕 

I think for the long term, we're _probably_ going to want to use mShots. I know in the past when people have asked about screenshot capturing, Systems has almost always just said "please use mShots." We're kind of playing with fire by relying on the free tier -- we wouldn't this tool to stop working partway through the testing because of an account limit. While the risk of anything bad is really, really low, we also don't know what thum.io does with all their generated screenshots.

But, those limitations you mentioned are certainly annoying... But, I think we can do a few things to mitigate them.

> I cannot find a way to capture the full page. The images were cropped especially for long pages. I used the param screen_height=3200 (this is the maximum) and the image was still cropped [here](https://s0.wp.com/mshots/v1/https%3A%2F%2Fwordpress.com%2Fsupport%2Fwriting-prompts?screen_height=3200).

Ack, it looks like they do this for performance reasons, and I guess that makes sense. We might be able to reach out to systems and see if they would be okay with some kind of exception to this.

I wonder though if those 3200 pixels might be enough for the testing purposes. That does let you look at a lot of the page! And we could just make the link at the top have `target="_blank"` so that it opens in a new tab. So if people want to take a deeper dive on a given page, they can open the full links to take a peek.

> If you enter a new URL for the first time, you need to wait and navigate to the URL a second time to see the generated image. I will then need to add a check to reload if the URL was redirected to https://s0.wp.com/mshots/v1/default

I think there's a few different things that we could do to help mitigate that! We could do any or all of these...
1. On csv load, we could just fire off all the mshot requests in the background so that they start the queueing process. That way, by the time they get to each URL, it's more likely to have already been saved off and cached. We could also limit this to URLs that don't have a status if we want.
2. We could add a manual "refresh" button that re-renders the preview components, so people can manually refresh to see if the image is saved off.
3. We could do what you mentioned by checking for redirects and not showing the images until the request stops redirecting. We could also do that for just the first URL and combine with option 1 above, so we effectively don't show the first comparison until we know that both a) we can pictures to show and b) all other URLs have been enqueued

Kind of unrelated, but one nice perk of mShots is that the urls can be used as `img` `src`s, so we could avoid some of the delayed repainting UI quirks that you get with `iframes`!

Anyway, I think what you have works for now, but that's just my 0.02! This is a hard tradeoff. 🤔

Reply:
> We're kind of playing with fire by relying on the free tier -- we wouldn't this tool to stop working partway through the testing because of an account limit.

Yeah, I am worried about this one! I have changed the API to use `mShots` instead of `Thum.io` 

> I wonder though if those 3200 pixels might be enough for the testing purposes. That does let you look at a lot of the page! And we could just make the link at the top have target="_blank" so that it opens in a new tab.

This makes sense! It doesn't necessarily have to be a full page all the time and 3200 pixels might be enough for most of the cases. I have added the `target="_blank"`. 

I have also moved all the related codes for the preview in [this file](https://github.com/Automattic/compare-snapshot-tool/pull/1/files#diff-8124db1530366ba95f3db6621463db43182264c1d014e27ce44350bf9300d67f).

> We could also limit this to URLs that don't have a status if we want.

This sounds like a good idea! mShots probably has a cache for the images for at least a day.

> We could add a manual "refresh" button that re-renders the preview components, so people can manually refresh to see if the image is saved off. 

I agree with this one too! 

> We could also do that for just the first URL and combine with option 1 above, so we effectively don't show the first comparison until we know that both a) we can pictures to show and b) all other URLs have been enqueued

This one seems to be the best option! I am mostly concern about the load that we will fire to mShots if the uploaded CSV has a lot of URLs. I am leaning towards the safer side! 

I really appreciate your fresh ideas here. I was a bit stuck on the implementation for this one!